pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -1253,10 +1253,21 @@ func (c *Reconciler) updateLabelsAndAnnotations(ctx context.Context, pr *v1beta1
 	return newPr, nil
 }
 
-func storePipelineSpecAndMergeMeta(pr *v1beta1.PipelineRun, ps *v1beta1.PipelineSpec, meta *metav1.ObjectMeta) error {
+func storePipelineSpecAndMergeMeta(pr *v1beta1.PipelineRun, ps *v1beta1.PipelineSpec, meta *tresources.ResolvedObjectMeta) error {
 	// Only store the PipelineSpec once, if it has never been set before.
 	if pr.Status.PipelineSpec == nil {
 		pr.Status.PipelineSpec = ps
+		if meta == nil {
+			return nil
+		}
+
+		// Propagate ConfigSource from remote resolution to PipelineRun Status
+		if meta.ConfigSource != nil {
```

Comment:
Is there a chance that, this might get populated in a reconcile after the first reconcile, in which case this code will not be reached because pr.Status.PipelineSpec will not be nil.

Reply:
Good point @jagathprakash! I guess this is something not impossible to happen but not sure how likely it will be. In normal cases, the source and the spec should be available from remote resolution at the same time. Agreed that we should avoid that if this can be the case.

What I did in the recent update is move out the configsource chunk out of the status.spec != nil check and also added tests for it. Please take a look and let me know what you think. Thank you!!